cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -236,3 +263,30 @@ def test_instance_attach_volume(new_instance, volume, appliance):
              delay=20,
              timeout=300, message="Waiting for volume to be attached to instance",
              fail_func=new_instance.refresh_relationships)
+
+
+@pytest.mark.rfe
+@pytest.mark.ignore_stream('5.9')
+def test_instance_attach_detach_volume_with_type(volume_with_type, new_instance, appliance):
+    initial_volume_count = new_instance.volume_count
+    new_instance.attach_volume(volume_with_type.name)
+    view = appliance.browser.create_view(navigator.get_class(new_instance, 'Details').VIEW)
+    view.flash.assert_success_message(
+        'Attaching Cloud Volume "{}" to {} finished'.format(volume_with_type.name, new_instance.name))
+
+    @wait_for_decorator(delay=10, timeout=300,
+                        message="Waiting for volume to be attached to instance")
+    def volume_attached_to_instance():
```

Comment:
Required:
I would like to shift this wait to `attach_volume` method with default argument `wait=False`.

`new_instance.attach_volume(volume_with_type.name, wait=True)` It should wait for volume attach. 
Similarly for `detach_volume` method.

Reply:
@digitronik Regarding Wait for attach and detach the volume, they was part of the methods but they were moved outside for the assert success to work. When you do the wait which performs refresh inside the method the flash message will disappear despite creating the View again